geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -432,34 +442,42 @@ protected void runDispatcher() {
           _messageQueue.remove();
           clientMessage = null;
         } catch (AuthenticationExpiredException expired) {
-          if (waitForReAuthenticationStartTime == -1) {
+          synchronized (reAuthenticationLock) {
```

Comment:
This should all be moved to a method invoked by this catch block. The new method should be unit testable.

Reply:
I did move them into a separate method for readability, but I didn't make it accessible for testing, because I would like it to remain as a private method. I did add a test for it though.

Reply:
Package private methods are acceptable for testing methods in isolation of the larger system.

Reply:
I already have unit tests covering this method from public accessors for both timeout and notified scenarios.